src/config.rs

```diff
@@ -46,6 +46,11 @@ pub struct Config {
     pub types: Option<Vec<LanguageType>>,
     // /// A map of individual language configuration.
     // pub languages: Option<HashMap<LanguageType, LanguageConfig>>,
+    /// Whether to output only the paths for downstream batch processing 
+    /// *Default:* false
+    #[serde(skip)]
+    /// adds a closure for each function, e.g., print the result
+    pub for_each_fn: Option<fn (LanguageType, String)>,
```

Comment:
Sorry for the late review, I've been busy with work, but this should return a `Report`, and the report JSON with the language is what should be printed, not just the file path.

Reply:
Will it be okay to have --streaming-json and --streaming as two separate options? Our batch processing would prefer a file at a line to simplify the bash downstreaming processes. If I generate JSON, it would require a json parser (an extra dependency on the jq tool).

Reply:
That sounds reasonable. I don't know about two separate options. I would instead change `--streaming` to accept an enum (see this example https://github.com/XAMPPRocky/tokei/blob/fd08c8b889f96986d3a05a17c58998ca062e9176/src/cli.rs#L135 ) with `Simple` and `Json` as its variants.

Reply:
I made the change, please see if you would like it and whether the enum Streaming should be moved to another module or be kept with the cli.rs module.

Reply:
@yijunyu Thank you again for working on this. The streaming enum seems fine where it is. I still think this should be changed from `fn (LanguageType, String)` to `fn (LanguageType, Report)`. Could we also change the simple format to include stats, I think people would be interested in that information. I think the simple format should look as follows.

```
{language} {path} {lines} {code} {comments} {blanks}

Rust src/main.rs 50 25 15 10
```

Reply:
Got this implemented on `fn <LanguageType, Report>`, easier than I initially thought, thanks to the predefined formatter.